static/js/components/forms/MenuItemForm.tsx

```diff
@@ -0,0 +1,182 @@
+import * as yup from "yup"
+
+import { InternalSortableMenuItem } from "../widgets/MenuField"
+import RelationField from "../widgets/RelationField"
+import { FormError } from "./FormError"
+
+import { LinkType } from "../../types/websites"
+
+interface Props {
+  onSubmit: (
+    values: any,
+    formikHelpers: FormikHelpers<any>
+  ) => void | Promise<any>
+  activeItem: InternalSortableMenuItem | null
+  collections?: string[]
+  existingMenuIds?: Set<string>
+}
+
+export type MenuItemFormValues = {
+  menuItemTitle: string
+  menuItemType: LinkType.Internal | LinkType.External
+  externalLink: string
+  internalLink: string
+}
+
+const schema = yup.object().shape({
+  menuItemTitle: yup
+    .string()
+    .required()
+    .label("Title"),
+  menuItemType: yup.mixed().oneOf([LinkType.Internal, LinkType.External]),
+  externalLink: yup
+    .string()
+    .label("External link")
+    .url()
+    .when("menuItemType", {
+      is:   LinkType.External,
+      then: yup.string().required()
+    }),
+  internalLink: yup
+    .string()
+    .label("Internal link")
+    .when("menuItemType", {
+      is:   LinkType.Internal,
+      then: yup.string().required()
+    })
+})
+
+const emptyInitialValues: MenuItemFormValues = {
+  menuItemTitle: "",
+  menuItemType:  LinkType.Internal,
+  externalLink:  "",
+  internalLink:  ""
+}
+
+export default function MenuItemForm({
+  onSubmit,
+  activeItem,
+  collections,
+  existingMenuIds
+}: Props): JSX.Element {
+  const initialValues = useMemo(
+    () =>
+      activeItem ?
+        {
+          menuItemTitle: activeItem.text,
+          menuItemType:  activeItem.targetUrl ?
+            LinkType.External :
+            LinkType.Internal,
+          externalLink: activeItem.targetUrl || "",
+          internalLink: activeItem.targetContentId || ""
+        } :
+        emptyInitialValues,
+    [activeItem]
+  )
+
+  return (
+    <Formik
+      onSubmit={onSubmit}
+      validationSchema={schema}
+      initialValues={initialValues}
+      enableReinitialize={true}
+    >
+      {({ isSubmitting, status, setFieldValue, values }) => {
+        return (
+          <Form className="row">
+            <div className="form-group w-100">
+              <label className="px-2" htmlFor="menuItemTitle">
+                Title
+              </label>
+              <Field
+                id="menuItemTitle"
+                name="menuItemTitle"
+                className="form-control"
+              />
+              <ErrorMessage name="menuItemTitle" component={FormError} />
+            </div>
+            <div className="form-group w-100">
+              <input
+                type="radio"
+                id="menuItemTypeInternal"
+                name="menuItemType"
+                value={LinkType.Internal}
+                checked={values.menuItemType === LinkType.Internal}
+                onChange={() => {
+                  setFieldValue("menuItemType", LinkType.Internal)
```

Comment:
It looks like this function and the similar one for external links aren't covered. Can we add a test that shows a change of state when these two radio buttons are clicked?